torch/csrc/jit/mobile/backport_manager.cpp

```diff
@@ -212,6 +213,35 @@ bool backport_v5_to_v4(
   return true;
 }
 
+bool backport_v6_to_v5(
```

Comment:
Q. Should we consider using an exception to indicate a failure to backport? Mentioning this since it is possible for someone to forget to check the return value of this method since the compiler can't check the `[[noreturn]]` annotation on functions called via pointer.

Reply:
In #58790, I add [a few check in the main while loop to catch backport failure](https://github.com/pytorch/pytorch/blob/03d1b0653fb905313c22e3e1e2a2a67140f309ef/torch/csrc/jit/mobile/backport_manager.cpp#L287-L328). Since this api is used in different model delivery infra and it's better to gracefully exit, I tend to use warning message + boolean to indicate the backport failure.

Reply:
imho, as long as the caller is able to catch the exception, it should be okay. In fact it maybe safer to use exceptions, since the risk of accidentally forgetting to check the return value goes away, and the call-site makes it super clear that an exception is expected. We could consider this as a future improvement to make the codebase safer from accidental errors resulting from not checking the return value (which is the intention behind the `[[noreturn]]` annotation in C++).

Reply:
I'm seeing this code in the link you referenced above:

<!--StartFragment-->
    // Keep backporting till request version
    std::stringstream backport_model_stream =
        bytecodeBackportFunctions()[bytecode_version--](input_model_stream);
<!--EndFragment-->

Has the API changed since then?